src/mjolnir/shortcutbuilder.cc

```diff
@@ -295,9 +295,11 @@ uint32_t ConnectEdges(GraphReader& reader,
   total_duration += edge_duration;
 
   // Add edge and turn duration for truck
-  // Currently use the same as for cars. TODO: implement for trucks
   total_truck_duration += turn_duration;
-  auto const truck_speed = tile->GetSpeed(directededge, kNoFlowMask, kInvalidSecondsOfWeek, true);
+  auto const truck_speed =
+      std::min(tile->GetSpeed(directededge, kNoFlowMask, kInvalidSecondsOfWeek, true),
+               directededge->truck_speed() ? directededge->truck_speed() : kMaxAssumedTruckSpeed);
```

Comment:
Truck speed is set on a directed edge only if there's a `maxspeed:hgv` tag on the way. If that is set, then don't clamp it.